backend/src/controller/plants.rs

```diff
@@ -16,19 +16,23 @@ use actix_web::{
 #[utoipa::path(
     context_path = "/api/plants",
     responses(
-        (status = 200, description = "Fetch or search for all plants by their common or species name", body = Vec<PlantsSummaryDto>)
+        (status = 200, description = "Fetch or search for all plants by their common or species name", body = PlantsSearchDto)
     )
 )]
 #[get("")]
 pub async fn find_all_or_search(
     query: Option<Query<PlantsSearchParameters>>,
     pool: Data<Pool>,
 ) -> Result<HttpResponse> {
-    let response = match query {
-        Some(parameters) => service::plants::search(&pool, &parameters).await?,
-        None => service::plants::find_all(&pool).await?,
+    let response = if let Some(parameters) = query {
+        let response = service::plants::search(&pool, &parameters).await?;
+        HttpResponse::Ok().json(response)
+    } else {
+        let response = service::plants::find_all(&pool).await?;
+        HttpResponse::Ok().json(response)
     };
-    Ok(HttpResponse::Ok().json(response))
+
+    Ok(response)
```

Comment:
This looks a little awkward which is due to differing types returned from `find_all` and `search`.
The generated openapi documentation is also wrong now.
I would go with a more generic pagination approach.
See other comment.